hikari/api/rest.py

```diff
@@ -3275,6 +3275,9 @@ async def add_user_to_guild(
             Requires the `MANAGE_NICKNAMES` permission on the guild.
         nick : hikari.undefined.UndefinedOr[builtins.str]
             Deprecated alias for `nickname`.
+
+            .. deprecated:: 2.0.0.dev106
+                Use `nickname` instead.
         roles : hikari.undefined.UndefinedOr[hikari.snowflakes.SnowflakeishSequence[hikari.guilds.PartialRole]]
             If provided, the roles to add to the user when he joins the guild.
             This may be a collection objects or IDs of existing roles.
```

Comment:
(ignore the code this is pointing to, just picked a random spot)
Can we also document special cases such as https://github.com/discord/discord-api-docs/pull/4872 and https://discord.com/developers/docs/interactions/application-commands#application-command-permissions-object-application-command-permissions-constants for the perm endpoints and ig also the perm object maybe?